static/js/pivotal.js

```diff
@@ -143,21 +173,78 @@
         'X-TrackerToken': pt_token
       },
       success: function(data) {
-        // Find all git commit messages
-        var list = data.filter(function(obj) {
-          return obj.commit_type;
-        });
-        // Find repo name from comment body text
-        list = list.map(function(item) {
-          return item.text.split('https://github.com/gengo/')[1].split('/commit/')[0];
+        var PULL_REQUEST_REGEX = /Merge pull request/;
+        var COMMIT_REPO_REGEX = /https:\/\/github.com\/gengo\/(.+)\/commit\//;
+        var DEPLOY_REPO_REGEX = new RegExp('Deployed (.+) to '+ config.environment +': ');
+
+        var activities = data.filter(function(activity) {
+          return activity.commit_type === 'github' || DEPLOY_REPO_REGEX.test(activity.text);
+        }).reverse();
+
+        var activitiesGrouped = groupBy(activities, function(activity) {
+          // With commit message
+          if (COMMIT_REPO_REGEX.test(activity.text)) {
+            return activity.text.match(COMMIT_REPO_REGEX)[1];
+          }
+          // Deployed message
+          if (DEPLOY_REPO_REGEX.test(activity.text)) {
+            return activity.text.match(DEPLOY_REPO_REGEX)[1];
+          }
         });
 
-        callback(removeDupesFromArray(list));
+        var inProgress = [];
+        var deployed = [];
+        var notDeployed = [];
```

Comment:
Maybe not so important, but I'm always wary of negatives. The brain plays tricks with them (like skipping the negation or struggling with double negatives).
What about `awaitingDeployment` or `readyToDeploy` or something ?